fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/security/KeyManager.java

```diff
@@ -88,6 +84,19 @@ public PoPToken getPoPToken(LaoView laoView, RollCall rollCall) throws KeyExcept
     return wallet.generatePoPToken(laoView.getId(), rollCall.getPersistentId());
   }
 
+  /**
+   * Generate a long-term PoP Token for the given Lao - ClientID pair
+   *
+   * @param laoId to generate the PoP Token from
+   * @param clientId to generate the PoP Token from
+   * @return the generated PoP Token
+   * @throws KeyGenerationException if an error occurs during key generation
+   * @throws UninitializedWalletException if the wallet is not initialized with a seed
+   */
+  public PoPToken getLongTermPoPToken(String laoId, String clientId) throws KeyException {
+    return wallet.generatePoPToken(laoId, Hash.hash(clientId));
+  }
```

Comment:
I have a nomenclature / typing problem here.
Is it really a `long term PoP token` or an entirely different object, say a `long term pseudonymous auth identity` ? And if so, shouldn't this have its own type (i.e., not `PoPToken`) ?

Reply:
Yes, that long term auth token object would also be a key pair essentially, but I agree it's cleaner having a proper type for it